packages/demand-api/src/loyalty/types.ts

```diff
@@ -4,20 +4,32 @@ export type ClientId = string // could be a list of supported clients but lib is
 
 export type LoyaltyStatusResponse = {
   balance?: number
-  can_burn_points?: boolean
+  can_burn?: boolean
+  can_earn?: boolean
 }
 
-export type MoneyToPointsParams = {
+export type BurnPointsCalcParams = {
   amount?: number
-  currency?: string // probably should be a list of supported currencies?
+  currency?: string
 }
 
-export type MoneyToPointsResponse = {
+export type BurnPointsCalcResponse = {
+  points?: number
+}
+
+export type EarnPointsCalcParams = {
+  total_amount?: number
+  burn_points?: number
+  currency?: string
+}
+
+export type EarnPointsCalcResponse = {
   points?: number
 }
 
 export type PreAuthParams = {
   points?: number
+  flexpay?: boolean
```

Comment:
These two are required

Reply:
Please mark `nonce` as required for `PreAuthResponse` as well (cannot comment on that line)

Reply:
`currency` is not needed in `/pre-auth` endpoint, it's just: ```POST /loyalty-<id>/pre-auth```

Reply:
currency is required in pre-auth endpoint

Reply:
right, I hadn't refreshed the confluence page 🤦

Reply:
but the currency should not be required, because _"if we don’t want to burn any points and be only eligible for earn, we should send the following"_
```Request
{
  "points": 0
  "flexpay": true,
}